### PR TITLE
Rework docker connection validation, so we can find out why docker connection failed.

### PR DESCRIPTION
**Pull request by Ihsahn**

### CODE REVIEW

**Comment by Ihsahn** [comment]
Build took to long so Travis decided to kill it, unfortunately I have no permission to restart it.

**Comment by neilcsmith-net** [comment]
@Ihsahn retriggered the failing test.

**Comment by Ihsahn** [comment]
> @Ihsahn retriggered the failing test.

thx!

**Comment by Ihsahn** [comment]
@matthiasblaesing can you take a look?

**Comment by matthiasblaesing** [comment]
If I remember correctly, log messages on level SEVERE raise an "exception" bubble in the IDE. While this might be helpful, I as a user would prefer to have the message in the place where I can do something about it. I think it would be good to see if the either the value `WizardDescriptor.PROP_ERROR_MESSAGE` can be enhanced or if another output channel is needed.

**Comment by Ihsahn** [comment]
@matthiasblaesing reworked. You were right about exception bubble and I was also mislead by logged exceptions (I thought that they come from my code, but apparently that was different logger...) - which I also fixed now.

File: ide/docker.api/src/org/netbeans/modules/docker/api/DockerAction.java [hunk 1]
```diff
@@ -1004,9 +1004,8 @@ public StreamItem fetch() {
         }
     }
 
-    public boolean ping() {
+    public boolean pingWithExceptions() throws IOException, MalformedURLException {
         assert !SwingUtilities.isEventDispatchThread() : "Remote access invoked from EDT";
-        try {
             Endpoint s = createEndpoint();
```
Comment (matthiasblaesing) [inline ide/docker.api/src/org/netbeans/modules/docker/api/DockerAction.java]: The idention is off - please move it to the correct level (one level to the left).

File: ide/docker.api/src/org/netbeans/modules/docker/api/DockerAction.java [hunk 2]
```diff
@@ -1004,9 +1004,8 @@ public StreamItem fetch() {
         }
     }
 
-    public boolean ping() {
+    public boolean pingWithExceptions() throws IOException, MalformedURLException {
```
Comment (matthiasblaesing) [inline ide/docker.api/src/org/netbeans/modules/docker/api/DockerAction.java]: I would not use such a precise `throws` clause. I would use `Exception` that way the code can evolve without being tied to the fixed API.

File: ide/docker.ui/src/org/netbeans/modules/docker/ui/wizard/DockerConnectionPanel.java [hunk 3]
```diff
@@ -246,14 +258,24 @@ public void validate() throws WizardValidationException {
             }
 
             DockerAction action = new DockerAction(instance);
-            if (!action.ping()) {
+            boolean pingResult = false;
+            
+            try {
+                pingResult = action.pingWithExceptions();
+            } catch (Exception ex) {
+                LOGGER.log(Level.FINE, "docker connection ping failed", ex); // NOI18N
```
Comment (matthiasblaesing) [inline ide/docker.ui/src/org/netbeans/modules/docker/ui/wizard/DockerConnectionPanel.java]: I would move this to `Level.INFO`. It is worth logging and INFO will not reach the UI, but in the log it is still useful.

**Comment by Ihsahn** [comment]
@matthiasblaesing Comments addressed

**Comment by matthiasblaesing** [comment]
Merged via https://github.com/apache/netbeans/commit/d5a1a036f6771916a326ec84025d0ad64a51b4a7 (sorry for some reason the author information was lost on rebase - I had to fix a wrong exception catch)

**Pull request closed without merging**